File: C++/Problems/Problems/problem_258.cpp
#include "problem_258.h"

bool problem_258::test() {
    auto num = 38;

    auto expected = 2;

    auto result = addDigits(num);

    std::cout << result << std::endl;

    return expected == result;
}

int problem_258::addDigits(int num) {
    if (num == 0)
        return 0;
    return 1 + (num - 1) % 9;
}

File: Python/Problems/Problems/problem258.py
from .problem import Problem


class Problem258(Problem):
    def test(self) -> bool:
        num = 38

        expected = 2

        result = self.addDigits(num)

        print(result)

        return expected == result

    def addDigits(self, num: int) -> int:
        if num == 0:
            return 0
        return 1 + (num - 1) % 9